### ISSUE TITLE
end-to-end ingress instructions

### ISSUE THREAD

**Issue by siggy** [linkerd/website#318]
## Background
https://linkerd.io/2/tasks/exposing-dashboard/ describes exposing `linkerd dashboard` via an ingress resource. Configuration of the ingress-controller (nginx or traefik) is left up to the user.

## Proposal

Update the docs to include an end-to-end working example of exposing `linkerd dashboard` via ingress, including ingress controller configs, service resources, and support for websockets.

These configs could even live in the https://github.com/linkerd/linkerd-examples repo, allowing the instructions to simply be:

```bash
linkerd install | kubectl apply -f -
curl https://raw.githubusercontent.com/linkerd/linkerd-examples/master/ingress/nginx.yml
open localhost.linkerd # or similar, prefer not requiring /etc/hosts modification
```

## Related issues

- linkerd/website#211
- linkerd/linkerd2#2143

**Comment by lzecca78** [comment]
Hi, we followed the doc here: https://linkerd.io/2/tasks/exposing-dashboard/  writing the following ingress : 

```
apiVersion: extensions/v1beta1
kind: Ingress
metadata:
  name: web-ingress
  namespace: linkerd
  annotations:
    kubernetes.io/ingress.class: internal
    nginx.ingress.kubernetes.io/configuration-snippet: |
      proxy_set_header l5d-dst-override linkerd-web.linkerd.svc.cluster.local:8084;
      proxy_set_header Origin "";
      proxy_hide_header l5d-remote-ip;
      proxy_hide_header l5d-server-id;
    nginx.ingress.kubernetes.io/auth-type: basic
    nginx.ingress.kubernetes.io/auth-secret: web-ingress-auth
    nginx.ingress.kubernetes.io/auth-realm: 'Authentication Required'
spec:
  rules:
    - host: 'linkerd.example.com'
      http:
        paths:
          - path: /
            backend:
              serviceName: linkerd-web
              servicePort: 8084
```
we can reach correctly the linkerd dashboard through the ingress, but for every websocket call, is raised an error. This is the error we see when we try to access to `/api/tap` : 
```
Bad Request
{"error":"websocket: not a websocket handshake: 'upgrade' token not found in 'Connection' header"}
```
with `linkerd dashboard` everything works fine. 
Does anyone else is experiencing a similar issue when exposing through an ingress?

**Comment by grampelberg** [comment]
@siggy what's missing from what is there now?

**Comment by grampelberg** [comment]
@lzecca78 sounds like you're running into issues with your load balancer provider. Are you using ELBs? I believe that ALBs are required for websockets in AWS.

**Comment by lzecca78** [comment]
@grampelberg yes, we are using ELB so far. I will try to convert them to ALB today. I will give here a feedback as soon as possible! Thanks for the tip!

**Comment by lzecca78** [comment]
@grampelberg  i can confirm  you that the issue was on the load balancer provider, once moved to ALBs, everything has started to work perfectly 👍

**Comment by wmorgan** [comment]
Nice. @grampelberg / @siggy anything we can do in `linkerd check`, or in the docs, to capture this?
